runtime/common/src/lib.rs

```diff
@@ -373,50 +372,136 @@ pub mod changes {
+			<T as pallet_investments::Config>::InvestmentId,
+			InvestmentPortfolio<Balance>,
+		>::new();
+
+		// Denote current tranche token balances before dry running collecting
+		orml_tokens::Accounts::<T>::iter_key_prefix(&investor).for_each(|currency| {
+			if let CurrencyId::Tranche(pool_id, tranche_id) = CurrencyId::from(currency) {
+				let balance = orml_tokens::Accounts::<T>::get(&investor, currency);
+				portfolio
+					.entry(TrancheCurrency::generate(pool_id, tranche_id))
+					.and_modify(|p| {
+						p.free_tranche_tokens = balance.free.into();
+						p.reserved_tranche_tokens = balance.reserved.into();
+					})
+					.or_insert(
+						InvestmentPortfolio::<Balance>::new()
+							.with_free_tranche_tokens(balance.free.into())
+							.with_reserved_tranche_tokens(balance.reserved.into()),
+					);
+			}
+		});
+
+		// Set pending invest currency and claimable tranche tokens
+		pallet_investments::InvestOrders::<T>::iter_key_prefix(&investor).for_each(|invest_id| {
+			// Collect such that we can determine claimable tranche tokens
+			// NOTE: Does not modify storage since RtAPI is readonly
+			let _ =
+				pallet_investments::Pallet::<T>::collect_investment(investor.clone(), invest_id);
+			let amount = pallet_investments::InvestOrders::<T>::get(&investor, invest_id)
+				.map(|order| order.amount())
+				.unwrap_or_default();
+			let free_tranche_tokens_new =
+				orml_tokens::Accounts::<T>::get(&investor, invest_id.into())
+					.free
+					.into();
+
+			portfolio
+				.entry(invest_id)
+				.and_modify(|p| {
+					p.pending_invest_currency = amount.into();
+					if p.free_tranche_tokens < free_tranche_tokens_new {
+						p.claimable_tranche_tokens =
+							free_tranche_tokens_new.saturating_sub(p.free_tranche_tokens);
+					}
+				})
+				.or_insert(
+					InvestmentPortfolio::<Balance>::new()
+						.with_pending_invest_currency(amount.into())
+						.with_claimable_tranche_tokens(free_tranche_tokens_new),
+				);
+		});
+
+		// Set pending tranche tokens and claimable invest currency
+		pallet_investments::RedeemOrders::<T>::iter_key_prefix(&investor).for_each(|invest_id| {
+			let pool_currency = PoolInspector::currency_for(invest_id.of_pool());
+			let balance_before: Balance = pool_currency
+				.map(|p_currency| {
+					orml_tokens::Accounts::<T>::get(&investor, p_currency)
+						.free
+						.into()
+				})
+				.unwrap_or_default();
+
+			// Collect such that we can determine claimable invest currency
+			// NOTE: Does not modify storage since RtAPI is readonly
+			let _ =
+				pallet_investments::Pallet::<T>::collect_redemption(investor.clone(), invest_id);
+			let amount = pallet_investments::RedeemOrders::<T>::get(&investor, invest_id)
+				.map(|order| order.amount())
+				.unwrap_or_default();
+			let balance_after: Balance = pool_currency
+				.map(|p_currency| {
+					orml_tokens::Accounts::<T>::get(&investor, p_currency)
+						.free
+						.into()
+				})
+				.unwrap_or_default();
```

Comment:
I think we should not make this dependent on `orml_tokens` Config. I would rather use the `fungibles::Inspect` and use restricted-tokens as the input in the runtimes.

Reply:
true! Thanks!